docs/markdown/Fs-module.md

```diff
@@ -29,3 +31,66 @@ name exists on the file system. This method follows symbolic links.
 
 Takes a single string argument and returns true if the path pointed to
 by the string is a symbolic link.
+
+## File Parameters
+
+### hash
+
+The `fs.hash(filename)` method computes the requested hash sum of a file.
+The available hash methods include: md5, sha1, sha224, sha256, sha384, sha512.
+
+### samefile
+
+The `fs.samefile(filename1, filename2)` method allows determining if two filenames refer to the same file.
+Perhaps a meson.build file in one place refer to a symlink and in another place a
+relative path and/or absolute path. The `samefile` method allows determining if these
+are the same file.
+
+Examples:
+
+```meson
+x = 'foo.txt'
+y = 'sub/../foo.txt'
+z = 'bar.txt'  # a symlink pointing to foo.txt
+
+fs.samefile(x, y)  # true
+fs.samefile(x, z)  # true
+```
+
+
+## Filename modification
+
+### with_suffix
+
+The `with_suffix` method is a *string manipulation* convenient for filename modifications.
+It allows changing the filename suffix like:
+
+## swap suffix
+
+```meson
+original = '/opt/foo.ini'
+new = fs.with_suffix('.txt')  # /opt/foo.txt
```

Comment:
All of these code samples take one argument, but the actual functions seem to take two.

Reply:
oops thanks